### ISSUE TITLE
Vue 3 V-model not working + Workaround

### ISSUE THREAD

**Issue by DjilanoS** [iamstevendao/vue-tel-input#270]
After trying to use this package I found that the Vue 3 version `vue-tel-input@next` is not working as intended.
V-model directive is not working due to the fact that the package still uses the old 'value' rather than emitting the new `modelValue` (https://github.com/iamstevendao/vue-tel-input/blob/master/src/components/vue-tel-input.vue#L94).

@iamstevendao Changes to the input and prop value is required in order to get it working with Vue 3.

```html
    <input
      ref="input"
      :type="inputOptions.type"
      :autocomplete="inputOptions.autocomplete"
      :autofocus="inputOptions.autofocus"
      :class="['vti__input', inputOptions.styleClasses]"
      :disabled="disabled"
      :id="inputOptions.id"
      :maxlength="inputOptions.maxlength"
      :name="inputOptions.name"
      :placeholder="parsedPlaceholder"
      :readonly="inputOptions.readonly"
      :required="inputOptions.required"
      :tabindex="inputOptions.tabindex"
      :value="modelValue"
      @blur="onBlur"
      @focus="onFocus"
      @input="$emit('update:modelValue', $event.target.value)"
      @keyup.enter="onEnter"
      @keyup.space="onSpace"
    />
```

```js
emits: ['update:modelValue'],

props: {
    modelValue: {
      type: String,
      default: '',
    },
}
```

### Temporary workaround for Vue 3 + Composition api.

```html
<vue-tel-input :value="phone" @input="onInput"></vue-tel-input>
```

```js
import { reactive, toRefs } from '@vue/reactivity'

export default {
    setup: () => {
          const state = reactive({
                phone: null
          }) 

        const onInput = (phone, phoneObject, input) => {
            if (phoneObject?.formatted) {
                state.phone = phoneObject.formatted
            }
        }

        return {
            ...toRefs(state),
            onInput
        }
    }
}
```

**Comment by iamstevendao** [comment]
@DjilanoS Thanks a lot! Would you mind creating a PR? Otherwise, I will update soon!

**Comment by DjilanoS** [comment]
@iamstevendao No problem. I wouldn't mind, although I'm not sure if this is everything that is needed in order to get it compatible as a plugin. Also noticed that the dropdown is also not working. Thanks for the reply!

**Comment by tsiory** [comment]
Hi @iamstevendao , any workaround to make the dropdown working on Vue 3 ?

**Comment by DjilanoS** [comment]
@tsiory You must be kidding me? the workaround is **LITERALLY** in the post.... perhaps you should read instead of replying to every issue with the same question.

**Comment by tsiory** [comment]
@DjilanoS I thought your issue was not related to dropdown behavior. Sorry if I make you angry. I'll try your workaround. Thanks for replying.

**Comment by LeonHarper** [comment]
@DjilanoS Thank you for your work and solution to this problem. But I do not believe it is fair to show frustration towards @tsiory as it is not clear that your initial solution applies to to drop down issue.

Initially you describe the problem as:

> V-model directive is not working due to the fact that the package still uses the old 'value' rather than emitting the new modelValue

to which you offer a solution.

A few days later you say:

> Also noticed that the dropdown is also not working. 

This appears to be a separate issue with no offered solution. I myself was watching this thread waiting for a solution to this second problem.

Again, thank you for you work. The solution is very much appreciated. But perhaps try to understand the position of others.

**Comment by DjilanoS** [comment]
> @DjilanoS Thank you for your work and solution to this problem. But I do not believe it is fair to show frustration towards @tsiory as it is not clear that your initial solution applies to to drop down issue.
> 
> Initially you describe the problem as:
> 
> > V-model directive is not working due to the fact that the package still uses the old 'value' rather than emitting the new modelValue
> 
> to which you offer a solution.
> 
> A few days later you say:
> 
> > Also noticed that the dropdown is also not working.
> 
> This appears to be a separate issue with no offered solution. I myself was watching this thread waiting for a solution to this second problem.
> 
> Again, thank you for you work. The solution is very much appreciated. But perhaps try to understand the position of others.

I think I'm in my rights to show my frustration when my email gets bombarded with the same question from the same person over and over again. Github issues does not exists to copy and paste your question over and over again. Then once again I told him to perhaps read. If he did, he would've noticed there's an answer to his question.. 

Apparently @iamstevendao is busy, I helped him a bit with a solution and a workaround for the community. No need to keep asking or pinging for solutions. Either find it yourself or patiently wait. 

Thanks.

**Comment by noeGnh** [comment]
For my part @DjilanoS solution didn't solve the dropdown issue, but using it with this [fork](https://github.com/victorybiz/vue3-tel-input), everything works fine. It can help until it's solved here.

**Comment by tsiory** [comment]
> For my part @DjilanoS solution didn't solve the dropdown issue, but using it with this [fork](https://github.com/victorybiz/vue3-tel-input), everything works fine. It can help until it's solved here.

Same for me. His workaround does not solve the dropdown issue. I'll try this fork tomorrow thanks for your help.

**Comment by DjilanoS** [comment]
Obviously it doesn't, the issue is not about the dropdown ;)

**Comment by iamstevendao** [comment]
Hey guys, sorry for the delay! I have addressed this issue in [v6.0.0-beta.1](https://github.com/iamstevendao/vue-tel-input/releases/tag/v6.0.0-beta.1).

Can anyone confirm that it's working now?

**Comment by reslear** [comment]
@iamstevendao yes thx bro, i'm tested, working good 👍
